server/src/main/java/org/elasticsearch/cluster/metadata/IndexAbstraction.java

```diff
@@ -257,4 +257,42 @@ private boolean isNonEmpty(List<IndexMetadata> idxMetas) {
             return (Objects.isNull(idxMetas) || idxMetas.isEmpty()) == false;
         }
     }
+
+    class DataStream implements IndexAbstraction {
+
+        private final org.elasticsearch.cluster.metadata.DataStream dataStream;
+        private final List<IndexMetadata> dataStreamIndices;
+        private final IndexMetadata writeIndex;
+
+        public DataStream(org.elasticsearch.cluster.metadata.DataStream dataStream,
+                          List<IndexMetadata> dataStreamIndices, IndexMetadata writeIndex) {
+            this.dataStream = dataStream;
+            this.dataStreamIndices = dataStreamIndices;
+            this.writeIndex = writeIndex;
```

Comment:
Yes, agreed. If that pr lands before this pr gets merged then I will adjust this pr and otherwise the pr that adds the generation should also adjust the code here.